<launch>
    <arg name="vehicle" default="typhoon_h480"/>

    <env name="GAZEBO_MODEL_PATH" value="${GAZEBO_MODEL_PATH}:$(find drcsim_model_resources)/gazebo_models"/>
    <env name="GAZEBO_RESOURCE_PATH" value="${GAZEBO_RESOURCE_PATH}:$(find drcsim_model_resources)/gazebo_models"/>
    <arg name="3d" value="true"/>
    <!-- Launch Gazebo simulation -->
	<rosparam param="/use_sim_time">true</rosparam>
    <node pkg="px4_bringup" type="launch_gzworld.py" name="gazebo_world" output="screen"
    args="-physics=ode -world=$(find multidrone_gazebo)/worlds/karting_sevilla.world
		  -add_model_path=$(find multidrone_gazebo)/models
		  -description_package=robots_description">
		<rosparam param="sim_origin">[37.558542, -5.931074, 7.89]</rosparam> 
		<!-- <rosparam param="sim_origin">[54.17435, 10.12996, 7.89]</rosparam>  -->
		<!-- <rosparam param="sim_origin">[52.4353142, 13.1424676, 7.89]</rosparam> -->
        <!-- [lat,lon,alt] -->
	</node>

	<!-- Launch robot id=1 -->
	<include file="$(find px4_bringup)/launch/spawn_robot.launch" ns="drone_2">
		<arg name="id" value="2"/>
		<arg name="material" value="Orange"/>
        <arg name="robot_model" value="typhoon_h480"/>
        <arg name="x" value="-5"/>
        <arg name="y" value="1"/>
        <arg name="z" value="0.0"/>
        <arg name="yaw" value="1.57"/>
	</include>

    <!-- Launch robot id=3  -->
    <include file="$(find px4_bringup)/launch/spawn_robot.launch" ns="drone_6">
        <arg name="id" value="6"/>
        <arg name="material" value="Indigo"/>
        <arg name="robot_model" value="typhoon_h480"/>
        <arg name="x" value="-51.75"/>
        <arg name="y" value="-28.32"/>
        <arg name="z" value="0"/>
        <arg name="yaw" value="3.141592"/>
    </include> 

       <!-- Launch robot id=2  -->
    <include file="$(find px4_bringup)/launch/spawn_robot.launch" ns="drone_4">
        <arg name="id" value="4"/>
        <arg name="material" value="Indigo"/>
        <arg name="robot_model" value="typhoon_h480"/>
        <arg name="x" value="-42.3241"/>
        <arg name="y" value="-1.2501"/>
        <arg name="z" value="0"/>
        <arg name="yaw" value="3.141592"/>
    </include>

    <!-- Launch UAL server for drone id=1 -->
    <node pkg="uav_abstraction_layer" type="server" name="ual" output="screen" ns="drone_2">
        <param name="uav_id" value="2"/>
        <param name="pose_frame_id" value="map"/>
        <rosparam param="home_pose">[-5, 1, 0.0]</rosparam>
        <rosparam param="map_origin_geo">[37.558542, -5.931074, 7.89]</rosparam>
        <!-- <rosparam param="map_origin_geo">[54.17435, 10.12996, 7.89]</rosparam> -->
        <!-- <rosparam param="map_origin_geo">[52.4353142, 13.1424676, 7.89]</rosparam> -->
    </node>



    <!-- Launch UAL server for drone id=2 -->
    <node pkg="uav_abstraction_layer" type="server" name="ual" output="screen" ns="drone_4">
        <param name="uav_id" value="4"/>
        <param name="pose_frame_id" value="map"/>
        <rosparam param="home_pose">[-42.32, -1.2501, 0.0]</rosparam>
        <rosparam param="map_origin_geo">[37.558542, -5.931074, 7.89]</rosparam>
        <!-- <rosparam param="map_origin_geo">[54.17435, 10.12996, 7.89]</rosparam> -->
        <!-- <rosparam param="map_origin_geo">[52.4353142, 13.1424676, 7.89]</rosparam> -->
    </node>
    
   <!-- Launch UAL server for drone id=5 -->
    <node pkg="uav_abstraction_layer" type="server" name="ual" output="screen" ns="drone_6">
        <param name="uav_id" value="6"/>
        <param name="pose_frame_id" value="map"/>
        <rosparam param="home_pose">[-51.75, -28.32, 0.0]</rosparam>
        <rosparam param="map_origin_geo">[37.558542, -5.931074, 7.89]</rosparam>
        <!-- <rosparam param="map_origin_geo">[54.17435, 10.12996, 7.89]</rosparam> -->
        <!-- <rosparam param="map_origin_geo">[52.4353142, 13.1424676, 7.89]</rosparam> -->
    </node>
    

    <include file="$(find global_tracker)/launch/global_tracker_sim.launch">
    </include>

    <!-- Start vehicle node --> 
    <node name="drc_vehicle_moving" pkg="multidrone_gazebo" type="drc_vehicle_moving" output="screen"/> 

    <group if="$(eval arg('3d') == true)">
		<include file="$(find global_tracker)/launch/global_3d_tracker.launch"/>
    </group>
</launch>